tools/nnyeah/nnyeah/Reworker.cs

```diff
@@ -278,6 +307,41 @@ void ReworkCodeBlock (MethodBody body)
 			}
 		}
 
+		static Instruction ChangeTypeInstruction (Instruction instruction, TypeDefinition typeDef)
+		{
+			if (instruction.Operand is TypeReference) {
+				return Instruction.Create (instruction.OpCode, typeDef);
+			}
+			// should never happen
+			throw new ArgumentException (nameof (instruction));
```

Comment:
Would InvalidOperationException be better for all of these?

Reply:
Unfortunately cecil uses object for the Operand property. :/